sympy/tensor/tensor.py

```diff
@@ -2624,6 +2638,8 @@ class Tensor(TensExpr):
 
     """
 
+    _diff_wrt = True
+    is_scalar = True
```

Comment:
Are these two necessary?

Reply:
Yes, they are, because if they are not added to tensor, the code complains that it does not know about how to derive tensors.

Reply:
That means you are passing somewhere through `Derivative`. I would recommend not to allow this, it's correct that the code complains, because you should use `PartialDerivative` instead of `Derivative`.

Reply:
Yeah, good point. It makes totally sense, to decouple these two. Within `_eval_partial_derivative` if there occurs a `Symbol` there is also `diff` invoked. Maybe there is some error occuring.

Reply:
Don't call `diff`.

Reply:
A tensor is not a scalar, this line is definitely wrong.

Reply:
Is there any possibility to perform the derivative by symbols without using `diff`?

Reply:
Yeah, I mentioned that at the beginning. Btw. a tensor of zeroth rank is scalar :-)

Reply:
Ok removed